framework/src/materials/Material.C

```diff
@@ -100,15 +102,25 @@ Material::Material(const InputParameters & parameters) :
 void
 Material::initStatefulProperties(unsigned int n_points)
 {
-  if (_has_stateful_property)
-    for (_qp = 0; _qp < n_points; ++_qp)
-      initQpStatefulProperties();
+  // checking for statefulness of properties via this loop is necessary
+  // because owned props might have been promoted to stateful by calls to
+  // getMaterialProperty[Old/Older] from other objects.  In these cases, this
+  // object won't otherwise know that it owns stateful properties.
+  for (auto& prop : _supplied_props)
```

Comment:
I am not sure but this could be not exactly. If the material `computQpProperties` is called in `FEProblemBase::initialSetup`, the old property should be properly initialized. However the material will not be evaluated on initial if there is no other objects depending on it. This investigation can leave to another PR, which hopefully will be in soon.